Using flag to adjust html based on the block:ellipsis or inline:no ellipsis needs

moses-v51 opened the pull request:
Moved onClick event up to span so that it can be reached by behavior
tests consistently

moses-v51 committed:
Moving onClick

Moved onClick event up to span so that it can be reached by behavior
tests consistently

moses-v51 committed:
style inline

Style is required to be inline on new div.  Reverted previous change as
I was able to get behavior tests to work without the change.

moses-v51 committed:
Using show ellipsis flag to determine html

Since the ellipsis requires a block element and other uses require an
inline element, making this adjust based on the need.

mikedaly commented:
I noticed the following:
- When `showEllipsis=true` and clicking on the control, the first character typed is getting lost.
- When `showEllipsis=true`, you can't use the tab key to navigation to the field, since the browser doesn't see any interactable element until you manually 'click'
- When `showEllipsis=false`, the width sizing isn't working correctly. The timezone is getting cutoff here:
  <img width="482" alt="screen shot 2016-07-05 at 1 10 19 pm" src="https://cloud.githubusercontent.com/assets/5348848/16598268/e21903de-42b1-11e6-8ce6-78543dd182cc.png">

moses-v51 committed:
Updates based on testing

Removed extra checkSelectAll as it is also called via another method.
For some reason the duplicate was resulting in the first letter entered
being removed.  Also changed min width back to inherit.  This will
cause the box-shadow to reappear, but allows the element to size
properly when being used in the date selector.

moses-v51 commented:
- When showEllipsis=true and clicking on the control, the first character typed is getting lost.

The checkSelectAll method was being called from both the onInputFocus and in onIsEditing.  Removing the duplicate call has prevented this behavior.
- When showEllipsis=true, you can't use the tab key to navigation to the field, since the browser doesn't see any interactable element until you manually 'click'

I don't know of a way around this.  Have you seen any?
- When showEllipsis=false, the width sizing isn't working correctly. The timezone is getting cutoff 

This was due to the minimum width being set to 100% so that the box-shadow would not visibly display and expand as you type.  I've tried other methods to remove the box-shadow, with no luck.  I've reset the minimum width back to inherit, as this issue is clearly more egregious.

moses-v51 committed:
Built previous change
